pennylane/resource/first_quantization.py

```diff
@@ -251,61 +310,119 @@ def norm(n, eta, omega, error, br=7, charge=0):
+            np.ceil(
+                np.log2(  # taken from Eq. (132) of PRX Quantum 2, 040332 (2021)
+                    (8 * np.pi * eta)
+                    / (error_uv * omega * bmin ** 2)
+                    * (eta - 1 + 2 * l_z)
+                    * (7 * 2 ** (n_p + 1) - 9 * n_p - 11 - 3 * 2 ** (-1 * n_p))
+                )
             )
         )
-        # computed using Eq. (113) of PRX Quantum 2, 040332 (2021)
-        lambda_nu_1 = lambda_nu + 4 / 2**n_m * (
-            7 * 2 ** (n_p + 1) - 9 * n_p - 11 - 3 * 2 ** (-1 * n_p)
-        )
-
-        p_nu = 0.2398  # approximation from Eq. (29) in arxiv:1807.09802
-        p_nu_amp = (
-            np.sin(3 * np.arcsin(np.sqrt(p_nu))) ** 2
-        )  # Eq. (129), PRX Quantum 2, 040332 (2021)
+        if not cubic:
+            BBt = np.matrix(recip_bv) @ np.matrix(recip_bv).T
+            is_orthogonal = (
+                np.linalg.norm(BBt - np.array([np.max(b ** 2) for b in recip_bv]) * np.identity(3))
+                < 1e-6
+            )
 
-        # lambda_u and lambda_v are taken from Eq. (25) of PRX Quantum 2, 040332 (2021)
-        lambda_u = eta * l_z * lambda_nu / (np.pi * omega ** (1 / 3))
-        lambda_v = eta * (eta - 1) * lambda_nu / (2 * np.pi * omega ** (1 / 3))
+        if not cubic and exact and n_p <= 6:
+            lambda_nu = 0
+            B_mus = {}
+            for j in range(2, n_p + 3):
+                B_mus[j] = []
+            for nu in product(range(-(2 ** (n_p)), 2 ** (n_p) + 1), repeat=3):
+                nu = np.array(nu)
+                if list(nu) != [0, 0, 0]:
+                    mu = int(np.floor(np.log2(np.max(abs(nu))))) + 2
+                    B_mus[mu].append(nu)
+            for mu in range(2, (n_p + 2)):
+                for nu in B_mus[mu]:
+                    Gnu_norm = np.linalg.norm(np.sum(nu * recip_bv, axis=0))
+                    lambda_nu += np.ceil(2 ** n_m * (bmin * 2 ** (mu - 2) / Gnu_norm) ** 2) / (
+                        2 ** n_m * (bmin * 2 ** (mu - 2)) ** 2
+                    )
+            lambda_nu_1 = lambda_nu
+        else:
+            lambda_nu = (  # expression is taken from Eq. (F6) of PRX 8, 011044 (2018)
+                4 * np.pi * (np.sqrt(3) * n ** (1 / 3) / 2 - 1)
+                + 3
+                - 3 / n ** (1 / 3)
+                + 3 * integrate.nquad(lambda x, y: 1 / (x ** 2 + y ** 2), [[1, n0], [1, n0]])[0]
+            ) / bmin ** 2
+            lambda_nu_1 = lambda_nu + 4 / (2 ** n_m * bmin ** 2) * (
+                7 * 2 ** (n_p + 1) - 9 * n_p - 11 - 3 * 2 ** (-1 * n_p)
+            )
 
-        # taken from Eq. (71) of PRX Quantum 2, 040332 (2021)
-        lambda_t_p = 6 * eta * np.pi**2 / omega ** (2 / 3) * 2 ** (2 * n_p - 2)
+        p_nu = lambda_nu_1 * bmin ** 2 / 2 ** (n_p + 6)
+        p_nu_amp, aa_steps = FirstQuantization._compute_AA_steps(p_nu, p_th)
 
+        # lambda_u and lambda_v are taken from Eq. (25) of PRX Quantum 2, 040332 (2021)
```

Comment:
Although they look changed, they are mostly identical to the ones before (esp. so for cubic lattices). Otherwise they are easy generalizations for noncubic lattices. Again, the same comment as the one above applies. The only eq. which I believe needs further clarification is `lambda_t_p` . I would just add a reference to Remark III.1 of [our work](https://arxiv.org/pdf/2302.07981.pdf) in line 378. I suggest

`# taken from Eq. (71) of PRX Quantum 2, 040332 (2021), applying Remark III.1 of arXiv:2302.07981`